### PR TITLE
test_immiscibleflash: convert to boost::test

### PR DESCRIPTION
**Pull request by akva2**
Sadly there is no support for templated fixtures so I have to do it manually.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: tests/material/test_immiscibleflash.cpp [hunk 1]
```diff
@@ -61,38 +62,30 @@ void checkSame(const FluidState& fsRef, const FluidState& fsFlash)
...
-                    << fsFlash.pressure(phaseIdx)  << " flash vs "
-                    << fsRef.pressure(phaseIdx) << " reference"
-                    << " error=" << error;
-                throw std::runtime_error(oss.str());
-        }
+        BOOST_CHECK_MESSAGE(std::abs(error) <= tol,
+                            "pressure error phase " + std::to_string(phaseIdx) +
+                            " is incorrect: " + std::to_string(fsFlash.pressure(phaseIdx)) +
+                            " flash vs " + std::to_string(fsRef.pressure(phaseIdx)) +
+                            " reference error=" + std::to_string(error));
```
Comment (bska) [inline tests/material/test_immiscibleflash.cpp:69]: My memory is hazy on the details, but I **vaguely** recall `BOOST_CHECK_MESSAGE` supporting streaming operators for string-like objects.  Are you able to write something like
```C++
BOOST_CHECK_MESSAGE(std::abs(error) <= tol,
"pressure error phase " << phaseIdx << " is incorrect: " << fsFlash.pressure(phaseIdx)
<< " flash vs " << fsRef.pressure(phaseIdx) << " reference error = " << error);
```
here?  Possibly at a cost of raising the minimum Boost version?

Comment (akva2) [reply tests/material/test_immiscibleflash.cpp]: indeed. updated.

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  This looks good to me now and I'll merge into master.

**Pull request merged**